rq/worker.py

```diff
@@ -1378,3 +1389,213 @@ class RandomWorker(Worker):
+        setup_loghandlers(logging_level, date_format, log_format)
+        self.register_birth()
+        self.log.info(
+            "ThreadPoolWorker %s: started with %s threads, version %s", self.key, self.threadpool_size, VERSION
+        )
+        self.log.warning("*** WARNING: ThreadPoolWorker is in beta and may be unstable. Don't use it in production!")
+        self.subscribe()
+        self.set_state(WorkerStatus.STARTED)
+        qnames = self.queue_names()
+        self.log.info('*** Listening on %s...', green(', '.join(qnames)))
+
+        if with_scheduler:
+            # Make scheduler ready.
+            pass
+
+        self._install_signal_handlers()
+        try:
+            while True:
+                try:
+                    self.check_for_suspension(burst)
+
+                    if self.should_run_maintenance_tasks:
+                        self.run_maintenance_tasks()
+
+                    if self._stop_requested:
+                        self.log.info('Worker %s: stopping on request', self.key)
+                        break
+
+                    if self.is_pool_full:
+                        self.log.debug('ThreadPool is full, waiting for idle threads...')
+                        self.__wait_for_slot()
+
+                    timeout = None if burst else self.dequeue_timeout
+                    result = self.dequeue_job_and_maintain_ttl(timeout)
+                    if result is None:
+                        if not burst:
+                            break
+                        has_pending_dependents = self.__check_pending_dependents()
+                        if has_pending_dependents:
+                            continue
+                        self.log.info("Worker %s: done, quitting", self.key)
+                        break
+
+                    job, queue = result
+                    self.reorder_queues(reference_queue=queue)
+                    self.execute_job(job, queue)
+                    self.heartbeat()
+
+                    completed_jobs += 1
+                    if max_jobs is not None:
+                        if completed_jobs < max_jobs:
+                            continue
+                        self.log.info("Worker %s: finished executing %d jobs, quitting", self.key, completed_jobs)
+                        break
+
+                except redis.exceptions.TimeoutError:
+                    self.log.error(f"Worker {self.key}: Redis connection timeout, quitting...")
+                    break
+
+                except StopRequested:
+                    break
+
+                except SystemExit:
+                    # Cold shutdown detected
+                    raise
+
+                except:  # noqa
+                    self.log.error('Worker %s: found an unhandled exception, quitting...', self.key, exc_info=True)
+                    break
+        finally:
+            self.register_death()
+            self.unsubscribe()
+
+        return bool(completed_jobs)
```

Comment:
@lowercase00 I'd like it so that you don't need to reimplement the whole `work()` method.

Do you mind opening a PR for a method called `worker.bootstrap()` that setups loghandlers to setting up scheduler? That way other workers can just implement the dequeue loop.

Depending on how this PR goes, I want to put in more hooks so that other workers can just implement already existing methods in the base class.

Reply:
Done